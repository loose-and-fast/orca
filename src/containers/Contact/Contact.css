.form {
    background-color: rgba(0, 0, 0, 0) !important
}

.button--submit {
  margin-top: 1rem !important;
}

.test {
  display: none;
}

.page_header {
  color: white !important;
}
