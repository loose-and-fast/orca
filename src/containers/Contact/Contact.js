import React, { Component } from 'react';
import axios from 'axios';
import {
  Grid,
  Header,
  Segment,
  Form,
  Button,
  TextArea,
  Loader,
  Dimmer,
  Message
} from 'semantic-ui-react';
import './Contact.css';
import PropTypes from 'prop-types';

class Contact extends Component {
  constructor(props) {
    super(props);
    this.state = {
      name: '',
      email: '',
      emailAddress: '',
      emailSubject: '',
      emailContent: '',
      loading: false,
      submitted: 'Not Submitted'
    };
  }

  handleChange(e, { name, value }) {
    this.setState({ [name]: value });
  }

  handleSubmit() {
    this.setState({ loading: true });
    axios
      .post('/contact', {
        test: this.state.email,
        email: this.state.emailAddress,
        name: this.state.name,
        subject: this.state.emailSubject,
        message: this.state.emailContent
      })
      .then(response => {
        this.setState({ loading: false, submitted: 'Successful' });
      })
      .catch(error => {
        this.setState({ loading: false, submitted: 'Unsuccessful' });
      });
    this.setState({
      name: '',
      emailAddress: '',
      emailSubject: '',
      emailContent: ''
    });
  }

  handleLoader() {
    if (this.state.loading) {
      return (
        <Dimmer active>
          <Loader>Loading</Loader>
        </Dimmer>
      );
    }
  }

  handleSuccess() {
    if (this.state.submitted === 'Successful') {
      return (
        <Message
          success
          header="Form Submitted"
          content="You have contacted Orca!"
        />
      );
    } else if (this.state.success === 'Unsuccessful') {
      return (
        <Message
          error
          header="Sorry, there was an error"
          content="Please refresh the page and try again."
        />
      );
    }
  }

  render() {
    const {
      name,
      email,
      emailAddress,
      emailSubject,
      emailContent
    } = this.state;
    return (
      <Grid.Row>
        {this.handleLoader()}
        <Header size="huge" className="page_header">
          contact us
        </Header>
        <Segment className="form" inverted>
          <Form onSubmit={this.handleSubmit}>
            <Form.Group widths="equal">
              <Form.Input
                required
                placeholder="Name"
                name="name"
                value={name}
                onChange={this.handleChange}
              />
              <Form.Input
                className="test"
                placeholder="Email"
                name="email"
                value={email}
                onChange={this.handleChange}
              />
              <Form.Input
                required
                placeholder="Email"
                name="emailAddress"
                type="email"
                value={emailAddress}
                onChange={this.handleChange}
              />
            </Form.Group>
            <Form.Input
              placeholder="Subject"
              name="emailSubject"
              value={emailSubject}
              onChange={this.handleChange}
            />
            <TextArea
              required
              autoHeight
              placeholder="What can we do for you?"
              name="emailContent"
              value={emailContent}
              onChange={this.handleChange}
              style={{
                minHeight: 200
              }}
            />
            {this.handleSuccess()}
            <Button className="button--submit" floated="left" type="submit">
              Submit
            </Button>
          </Form>
        </Segment>
      </Grid.Row>
    );
  }
}

Contact.propTypes = {};

export default Contact;
