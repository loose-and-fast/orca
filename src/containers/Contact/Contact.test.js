import React from 'react'
import { shallow } from 'enzyme';

import Contact from './Contact'

describe('Contact', () => {
  let component, props

  beforeEach(() => {
    props = {}
    component = shallow(<Contact {...props} />)
  })

  it('should', () => {
    expect(component).toMatchSnapshot()
  })
})