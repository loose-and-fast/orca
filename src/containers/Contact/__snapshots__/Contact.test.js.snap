// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact should 1`] = `undefined`;
